core/storage/merkle/polkadot_trie_db/polkadot_trie_db.cpp

```diff
@@ -37,8 +37,7 @@ namespace kagome::storage::merkle {
 
   PolkadotTrieDb::PolkadotTrieDb(std::unique_ptr<PersistentBufferMap> db,
                                  std::shared_ptr<hash::Hasher> hasher)
```

Comment:
I don't think you need to depend on hasher, because you already have `PolkadotCodec`, which has `hash256` method - and it can be mocked. Why do you use it here otherwise?